### ISSUE TITLE
GKL  failures in MacOS

### ISSUE THREAD

**Issue by magicDGS** [broadinstitute/gatk#3532]
## System

* GATK4 a1eee32e84c21c2f265d248c5f47789ae0ba2b37
* Mac OS X 10.11.6 x86_64
* machdep.cpu.features: FPU VME DE PSE TSC MSR PAE MCE CX8 APIC SEP MTRR PGE MCA CMOV PAT PSE36 CLFSH DS ACPI MMX FXSR SSE SSE2 SS HTT TM PBE SSE3 DTES64 MON DSCPL VMX SMX EST TM2 SSSE3 CX16 TPR PDCM SSE4.1 SSE4.2 POPCNT
* machdep.cpu.extfeatures: SYSCALL XD EM64T LAHF RDTSCP TSCI
* java version "1.8.0_60"
* Java(TM) SE Runtime Environment (build 1.8.0_60-b27)
* Java HotSpot(TM) 64-Bit Server VM (build 25.60-b23, mixed mode)

## Error

When updating a downstream project with the latest master and running the integration/unit tests with gradle, it generates the following error.

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGILL (0x4) at pc=0x00000001236427f4, pid=4010, tid=20739
#
# JRE version: Java(TM) SE Runtime Environment (8.0_60-b27) (build 1.8.0_60-b27)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.60-b23 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [libgkl_compression7227189416687158431.dylib+0x17f4]  Java_com_intel_gkl_compression_IntelDeflater_resetNative+0x164
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/daniel/workspaces/ReadTools/hs_err_pid4010.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

Error report file: [hs_err_pid4010.log.txt](https://github.com/broadinstitute/gatk/files/1259963/hs_err_pid4010.log.txt)

## Forcing other GKL versions

* 0.5.2 (working)
* 0.5.3 (failing)
* 0.5.5 (failing)
* 0.5.6 (failing)
* 0.5.7 (failing)
* 0.5.8 (failing)

**Comment by cmnbroad** [comment]
@erniebrau Can you take a look ?

**Comment by erniebrau** [comment]
Yes, we will take a look.

**Comment by erniebrau** [comment]
@magicDGS Does it fail in the same way when using GKL 0.5.3-0.5.8?

**Comment by magicDGS** [comment]
@erniebrau, the mesage and log file corresponds to the latest master (GKL 0.5.8); for the GKL 0.5.3 the error message is the following:

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGILL (0x4) at pc=0x000000011dc557f4, pid=20586, tid=20739
#
# JRE version: Java(TM) SE Runtime Environment (8.0_60-b27) (build 1.8.0_60-b27)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.60-b23 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# C  [libgkl_compression1417468606951982528.dylib+0x17f4]  Java_com_intel_gkl_compression_IntelDeflater_resetNative+0x164
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /Users/daniel/workspaces/ReadTools/hs_err_pid20586.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
```

And the log file: [hs_err_pid20586.log.txt](https://github.com/broadinstitute/gatk/files/1264191/hs_err_pid20586.log.txt)

Let me know if you need more information.

**Comment by magicDGS** [comment]
Any progress on this, @erniebrau? I would like to update to the latest master my downstream project... Thanks in advance!

**Comment by erniebrau** [comment]
@magicDGS Not yet. Apologies for the long delay. We've had some other urgent issues and features we're working on. I will make some time to look at this this week. I might be asking you some more questions about this during the debug process, since we do not currently have a machine (non-AVX Mac) where we can reproduce the issue.

**Comment by magicDGS** [comment]
Thanks for the update, @erniebrau - I would answer every question as soon as I read it. Thank you for doing this!

**Comment by erniebrau** [comment]
@magicDGS Can you please compile (with a simple `g++ avx-all.cpp`) and run the short program below on the affected Mac and report  back the output? This is basically the code being executed by GKL to determine if AVX is supported.

Code to compile and run (zipped): [code.zip](https://github.com/broadinstitute/gatk/files/1310975/code.zip)

**Comment by magicDGS** [comment]
The output for running the `a.out` is:

```
This CPU does NOT support AVX
This CPU supports AVX-512
```

Thanks for looking at this @erniebrau - let me know if you need something else...

**Comment by erniebrau** [comment]
@magicDGS We have an idea of what the problem is. If we're right, it will be fixed in the next GKL release, which should happen sometime tomorrow. I will then create a GATK branch that uses the new GKL release for you to test (and to ultimately merge into master).

If we're wrong about what is causing the problem, we will have to continue to debug. Sorry about the slowness of this process. Like I said before, it's mostly due to us not having a system where we can reproduce the issue.

**Comment by magicDGS** [comment]
Thanks @erniebrau! Looking forward to test the next release of GKL.

**Comment by erniebrau** [comment]
@magicDGS Just want to keep you up-to-date. The release is taking a bit longer than expected. I'll let you know when it's out so you can test it out.

**Comment by magicDGS** [comment]
Thanks @erniebrau - with GKL 0.7 I do not experience any failure anymore!

**Comment by erniebrau** [comment]
Excellent! 👍
